src/container-mapping.ts

```diff
@@ -0,0 +1,184 @@
+import { CommandType, Constants, getEncodedContent, writeToOutStream } from "./msdo-helpers";
+import { IMicrosoftSecurityDevOps } from "./msdo-interface";
+import * as https from "https";
+import * as core from '@actions/core';
+import * as exec from '@actions/exec';
+
+const sendReportRetryCount: number = 1;
+
+/**
+ * Represents the tasks for container mapping that are used to fetch Docker images pushed in a job run.
+ */
+export class ContainerMapping implements IMicrosoftSecurityDevOps {
+    readonly succeedOnError: boolean;
+
+    constructor() {
+        this.succeedOnError = true;
+    }
+
+    /**
+     * Container mapping pre-job commands wrapped in exception handling.
+     */
+    public runPreJob() {
+        try {
+            writeToOutStream("::group::Microsoft Defender for DevOps container mapping pre-job - https://go.microsoft.com/fwlink/?linkid=2231419");
+            this._runPreJob();
+        }
+        catch (error) {
+            // Log the error
+            writeToOutStream("Error in Container Mapping pre-job: " + error);
+        }
+        finally {
+            // End the collapsible section
+            writeToOutStream("::endgroup::");
+        }
+    }
+
+    
+    /*
+    * Set the start time of the job run.
+    */
+    private _runPreJob() {
+        const startTime = new Date().toISOString();
+        core.saveState('PreJobStartTime', startTime);
+        console.log('PreJobStartTime', startTime);
+    }
+
+    /**
+     * Placeholder / interface satisfier for main operations
+     */
+    public async runMain() {
+        // No commands
+    }
+
+    /**
+     * Container mapping post-job commands wrapped in exception handling.
+     */
+    public async runPostJob() {
+        try {
+            writeToOutStream("::group::Microsoft Defender for DevOps container mapping post-job - https://go.microsoft.com/fwlink/?linkid=2231419");
+            await this._runPostJob();
+        }
+        catch (error) {
+            // Log the error
+            writeToOutStream("Error in Container Mapping post-job: " + error);
+        }
+        finally {
+            // End the collapsible section
+            writeToOutStream("::endgroup::");
+        }
+    }
+
+    /*
+    * Using the start time, fetch the docker events and docker images in this job run and log the encoded output
+    * Send the report to Defender for DevOps
+    */
+    private async _runPostJob() {
+        let startTime = core.getState('PreJobStartTime');
+        if (startTime.length <= 0) {
+            startTime = new Date(new Date().getTime() - 10000).toISOString();
+            console.log(`PreJobStartTime not defined, using now-10secs`);
+        }
+        console.log(`PreJobStartTime: ${startTime}`);
+
+        let reportData = {
+            dockerVer: "",
+            dockerEvents: "",
+            dockerImages: ""
+        };
+        
+        // Initialize the commands 
+        await exec.exec('docker --version', null, {

```

Comment:
optimizations for later - running these commands async (like we do in ADO) to reduce the runtime even further